boot/zephyr/nrf_cleanup.c

```diff
@@ -40,10 +48,28 @@ void nrf_cleanup_peripheral(void)
 #if defined(NRF_UARTE0)
     nrf_uarte_disable(NRF_UARTE0);
     nrf_uarte_int_disable(NRF_UARTE0, 0xFFFFFFFF);
+#if defined(NRF_DPPIC)
+    /* Clear all SUBSCRIBE configurations. */
+    memset((uint8_t *)NRF_UARTE0 + 0x80, 0, 0x80);
```

Comment:
Since we are assuming the location of SUBSCRIBE and PUBLISH configuration is the same irrespective of the target SoC, I suggest we future proof this code with sufficient BUILD_ASSERTS. For example, we can guard this memset by the following two BUILD_ASSERTs. Similar BUILD_ASSERTs can be added for the other memsets too.



    BUILD_ASSERT((uint8_t *)NRF_UARTE0 + 0x80 == (uint8_t*)&NRF_UARTE0->SUBSCRIBE_STARTRX,
        "Location of UART subscribe configuration is not known for current SoC.");
    BUILD_ASSERT((uint8_t *)NRF_UARTE0 + 0x80 + 0x80 == (uint8_t*)&NRF_UARTE0->EVENTS_CTS,
        "Location of UART subscribe configuration is not known for current SoC.");

I also suggest using macros instead of magic numbers here. 
eg) 
`
#define SUBSCRIBE_CONFIG_OFFSET 0x80 /* Offset of the memory location of the first SUBSCRIBE register. */ 

#define SUBSCRIBE_CONFIG_SIZE 0x80 /* Total memory size occupied by all SUBSCRIBE registers of the UART peripheral. */
`

Reply:
better now?

Reply:
yes. Thanks.